apps/storefront/src/pages/Dashboard/index.tsx

```diff
@@ -224,6 +249,12 @@ function Dashboard(props: PageProps) {
           }}
         />
       </Box>
+      <ConfirmMasqueradeDialog
+        isOpen={Boolean(cartEntityId)}
+        isRequestLoading={isRequestLoading}
+        handleClose={() => setCartEntityId('')}
+        handleConfirm={() => startActing(tempCompanyId)}
```

Comment:
```tsx
handleConfirm={
  async () => {
    if (confirmMasquerade) { // this `if` is only here to please TS 🤷 
      await startActing(confirmMasquerade.companyId);
      setConfirmMasquerade(undefined);
    }
  }
}
```

Reply:
sounds like a case for an `asserts` then?

```tsx
function assertDefined<T>(value?: T): asserts value is T {
  if (value === undefined) {
    throw new Error("Expected value to be defined, but got undefined");
  }
}
```

useful when the code forces us to explain to Typescript that a scenario is technically impossible e.g.
this modal being open and yet `confirmMasquerade` equal to `undefined`

definitely a technique of last resort, but it'd take some epic refactoring of this Dialogue component for Typescript to work out this scenario is impossible on it's own 🍹